lib/pbench/agent/tool_data_sink.py

```diff
@@ -398,10 +457,13 @@ def execute(self):
                         self.state_change(data)
         except self.Terminate as exc:
```

Comment:
We could consider dropping the `Terminate` exception in favor of unsubscribing from the channel and handling without a warning issue at line 430.

Reply:
Done.